ground/gcs/src/plugins/config/configautotunewidget.cpp

```diff
@@ -85,127 +117,128 @@ void ConfigAutotuneWidget::saveStabilization()
 {
-    forumInteractionForm = new Utils::ForumInteractionForm(this);
-    connect(forumInteractionForm, SIGNAL(finished(int)), this, SLOT(onForumInteractionSet(int)));
-    ExtensionSystem::PluginManager *pm=ExtensionSystem::PluginManager::instance();
-    Core::Internal::GeneralSettings * settings=pm->getObject<Core::Internal::GeneralSettings>();
-    if (!settings->getForumPassword().isEmpty()) {
-       forumInteractionForm->setPassword(settings->getForumPassword());
-       forumInteractionForm->setUserName(settings->getForumUser());
+    autotuneShareForm = new AutotuneShareForm(this);
+    connect(autotuneShareForm, SIGNAL(finished(int)), this, SLOT(onShareFinished(int)));
+    connect(autotuneShareForm, SIGNAL(ClipboardRequest()), this, SLOT(onShareToClipboard()));
+    connect(autotuneShareForm, SIGNAL(DatabaseRequest()), this, SLOT(onShareToDatabase()));
+
+    QString currentFrameType;
+    SystemSettings *sysSettings = SystemSettings::GetInstance(getObjectManager());
+    if(sysSettings) {
+        UAVObjectField *frameType = sysSettings->getField("AirframeType");
+        if(frameType) {
+            autotuneShareForm->setVehicleTypeOptions(frameType->getOptions());
+            currentFrameType = frameType->getValue().toString();
+        }
     }
-    forumInteractionForm->setObservations(settings->getObservations());
-    forumInteractionForm->setAircraftDescription(settings->getAircraftDescription());
-    forumInteractionForm->show();
-    forumInteractionForm->raise();
-    forumInteractionForm->activateWindow();
+    // would be nice to set board type dropdown too but too hard to get a list (for now)
+
+    // fetch last shared data
+    loadUserData();
+
+    // we will overwrite some things though
+    Core::IBoardType *board = utilMngr->getBoardType();
+    if(board) {
+        autotuneShareForm->setBoardType(board->shortName());
+        autotuneShareForm->disableBoardType(true);
+    }
+    if(!currentFrameType.isNull()) {
+        autotuneShareForm->setVehicleType(currentFrameType);
+        autotuneShareForm->disableVehicleType(true);
+    }
+
+    autotuneShareForm->show();
+    autotuneShareForm->raise();
+    autotuneShareForm->activateWindow();
 }
 
-void ConfigAutotuneWidget::onForumInteractionSet(int value)
+void ConfigAutotuneWidget::onShareFinished(int value)
 {
-    if (!value) {
-        forumInteractionForm->deleteLater();
-        return;
-    }
-    ExtensionSystem::PluginManager *pm=ExtensionSystem::PluginManager::instance();
-    Core::Internal::GeneralSettings * settings=pm->getObject<Core::Internal::GeneralSettings>();
-    if (forumInteractionForm->getSaveCredentials()) {
-        settings->setForumPassword(forumInteractionForm->getPassword());
-        settings->setForumUser(forumInteractionForm->getUserName());
-    } else {
-        settings->setForumPassword("");
-        settings->setForumUser("");
+    Q_UNUSED(value);
+    autotuneShareForm->deleteLater();
```

Comment:
You can also create a single instance of the form on the contructor and only show/close when needed.
One other way is to use the Qt::WA_DeleteOnClose attribute on the form dialog.